Billing.API/Controllers/InvoiceController.cs

```diff
@@ -40,10 +44,10 @@ public InvoiceController(ILogger<InvoiceController> logger, IInvoiceService invo
             return Ok(response);
         }
 
+        [AllowAnonymous]
         [HttpGet]
-        [Route("/accounts/{origin}/{clientId:int:min(1)}/invoice/{filename}")] //Deprecated route
         [Route("/accounts/{origin}/{clientId:int:min(1)}/invoices/{filename}")]
-        public async Task<IActionResult> GetInvoiceFile([FromRoute] string origin, [FromRoute] int clientId, [FromRoute] string filename)
+        public async Task<IActionResult> GetInvoiceFilePublic([FromRoute] string origin, [FromRoute] int clientId, [FromRoute] string filename, [FromQuery] string signature)
```

Comment:
Yo dejaría el nombre anterior

```suggestion
        public async Task<IActionResult> GetInvoiceFile([FromRoute] string origin, [FromRoute] int clientId, [FromRoute] string filename, [FromQuery] string signature)
```